### PR TITLE
Remove product validation in ScanSettingBinding

### PR DESCRIPTION
**Pull request by Vincent056**
This commit removes product validation in ScanSettingBinding so we can launch both `rhcos4` and `ocp4-node` scan in one SSB.

ex.

```
apiVersion: compliance.openshift.io/v1alpha1
kind: ScanSettingBinding
metadata:
  name: ocp4-rhcos4-moderate-mix
  namespace: openshift-compliance
profiles:
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: ocp4-moderate-node
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: rhcos4-moderate
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: ocp4-moderate
settingsRef:
  apiGroup: compliance.openshift.io/v1alpha1
  kind: ScanSetting
  name: default
```

### CODE REVIEW

**Review by JAORMX — COMMENTED** [review]
Is this something you really wanna do? I recall us having limitations on the scans we can launch from a specific scan setting binding, and this might become problematic when scheduling them

**Comment by Vincent056** [comment]
> Is this something you really wanna do? I recall us having limitations on the scans we can launch from a specific scan setting binding, and this might become problematic when scheduling them

hi @JAORMX it's good to hear from you! I wonder if you still remember what the limitations were? I spent some time digging into the code, but couldn't figure out why it was designed this way. 

And I also did some testing to run scans with
```
apiVersion: compliance.openshift.io/v1alpha1
kind: ScanSettingBinding
metadata:
  name: ocp4-rhcos4-moderate-mix
  namespace: openshift-compliance
profiles:
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: ocp4-moderate-node
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: rhcos4-moderate
- apiGroup: compliance.openshift.io/v1alpha1
  kind: Profile
  name: ocp4-moderate
settingsRef:
  apiGroup: compliance.openshift.io/v1alpha1
  kind: ScanSetting
  name: default
```
All the scans were launched and working as expected, tho I will do more testing on it before making this change.

**Review by JAORMX — APPROVED** [review]

**Comment by Vincent056** [comment]
we are affecting by https://github.com/ComplianceAsCode/content/commit/6343659d1d25e97c66a5c1eaf8eb2ee20d1af920:
```
[vincent@node compliance-operator]$ for i in `oc get node|grep -i master|cut -d " " -f1`; do echo $i;oc debug -q node/$i -- ls -l /host/var/log/kube-apiserver;done;
ip-10-0-157-230.ec2.internal
Warning: would violate PodSecurity "restricted:latest": host namespaces (hostNetwork=true, hostPID=true), privileged (container "container-00" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "container-00" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "container-00" must set securityContext.capabilities.drop=["ALL"]), restricted volume types (volume "host" uses restricted volume type "hostPath"), runAsNonRoot != true (pod or container "container-00" must set securityContext.runAsNonRoot=true), runAsUser=0 (container "container-00" must not set runAsUser=0), seccompProfile (pod or container "container-00" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
total 130948
-rw-------. 1 root root 79922585 Jan 16 09:06 audit.log
-rw-r--r--. 1 root root        4 Jan 16 08:39 termination.log
ip-10-0-188-243.ec2.internal
Warning: would violate PodSecurity "restricted:latest": host namespaces (hostNetwork=true, hostPID=true), privileged (container "container-00" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "container-00" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "container-00" must set securityContext.capabilities.drop=["ALL"]), restricted volume types (volume "host" uses restricted volume type "hostPath"), runAsNonRoot != true (pod or container "container-00" must set securityContext.runAsNonRoot=true), runAsUser=0 (container "container-00" must not set runAsUser=0), seccompProfile (pod or container "container-00" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
total 106436
-rw-------. 1 root root 104856885 Jan 16 09:05 audit-2024-01-16T09-05-41.016.log
-rw-------. 1 root root   2514143 Jan 16 09:06 audit.log
-rw-r--r--. 1 root root         4 Jan 16 08:37 termination.log
ip-10-0-199-110.ec2.internal
Warning: would violate PodSecurity "restricted:latest": host namespaces (hostNetwork=true, hostPID=true), privileged (container "container-00" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "container-00" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "container-00" must set securityContext.capabilities.drop=["ALL"]), restricted volume types (volume "host" uses restricted volume type "hostPath"), runAsNonRoot != true (pod or container "container-00" must set securityContext.runAsNonRoot=true), runAsUser=0 (container "container-00" must not set runAsUser=0), seccompProfile (pod or container "container-00" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
total 117908
-rw-------. 1 root root 103748183 Jan 16 09:06 audit.log
-rw-------. 1 root root   1132900 Jan 16 08:43 termination.log
```

We might need to update the OCP version:
```
[vincent@node compliance-operator]$ oc get clusterversion
NAME      VERSION                                                    AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.12.0-0.ci.test-2024-01-16-081659-ci-op-v3mjd6zh-latest   True        False         25m     Cluster version is 4.12.0-0.ci.test-2024-01-16-081659-ci-op-v3mjd6zh-latest
```

**Comment by xiaojiey** [comment]
Per https://issues.redhat.com/browse/OCPBUGS-11856, seems it was fixed on 4.14 and higher versions

File: tests/e2e/serial/main_test.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func TestSuiteScan(t *testing.T) {
 				},
 			},
 			ID:       "xccdf_org.ssgproject.content_rule_coreos_vsyscall_kernel_argument",
-			Status:   compv1alpha1.CheckResultInfo,
+			Status:   compv1alpha1.CheckResultFail,
```
Comment (rhmdnd) [inline tests/e2e/serial/main_test.go:235]: Does this need to be pulled out into it's own PR since https://github.com/ComplianceAsCode/content/pull/11329 landed?

Comment (Vincent056) [reply tests/e2e/serial/main_test.go:235]: https://github.com/ComplianceAsCode/compliance-operator/pull/490 create a PR here

**Comment by xiaojiey** [comment]
@Vincent056 I retested the PR, I still got the same result for moderate profiles.
After all auto-remediations applied, the test suite showed INCONSISTENT, and some rules showed FAIL even if the auto-remediation are available for them.
```
$ oc get ccr -l compliance.openshift.io/automated-remediation=,compliance.openshift.io/check-status=FAIL
NAME                                                                   STATUS   SEVERITY
rhcos4-moderate-master-service-debug-shell-disabled                    FAIL     medium
rhcos4-moderate-master-sysctl-net-core-bpf-jit-harden                  FAIL     medium
rhcos4-moderate-master-sysctl-net-ipv6-conf-all-accept-ra              FAIL     medium
rhcos4-moderate-master-sysctl-net-ipv6-conf-all-accept-redirects       FAIL     medium
rhcos4-moderate-master-sysctl-net-ipv6-conf-default-accept-ra          FAIL     medium
rhcos4-moderate-master-sysctl-net-ipv6-conf-default-accept-redirects   FAIL     medium
rhcos4-moderate-worker-service-debug-shell-disabled                    FAIL     medium
rhcos4-moderate-worker-sysctl-net-core-bpf-jit-harden                  FAIL     medium
rhcos4-moderate-worker-sysctl-net-ipv6-conf-all-accept-ra              FAIL     medium
rhcos4-moderate-worker-sysctl-net-ipv6-conf-all-accept-redirects       FAIL     medium
rhcos4-moderate-worker-sysctl-net-ipv6-conf-default-accept-ra          FAIL     medium
rhcos4-moderate-worker-sysctl-net-ipv6-conf-default-accept-redirects   FAIL     medium
$ oc get ccr -l compliance.openshift.io/inconsistent-check
NAME                                                                               STATUS         SEVERITY
ocp4-moderate-node-worker-kubelet-anonymous-auth                                   INCONSISTENT   medium
ocp4-moderate-node-worker-kubelet-eviction-thresholds-set-hard-imagefs-available   INCONSISTENT   medium
ocp4-moderate-node-worker-kubelet-eviction-thresholds-set-hard-memory-available    INCONSISTENT   medium
ocp4-moderate-node-worker-kubelet-eviction-thresholds-set-hard-nodefs-available    INCONSISTENT   medium
ocp4-moderate-node-worker-kubelet-eviction-thresholds-set-hard-nodefs-inodesfree   INCONSISTENT   medium

```

**Comment by BhargaviGudi** [comment]
Verification failed with 4.15.0-0.nightly-2024-02-12-213938 + compliance-operator from PR #489 

Verification failed for moderate profiles.
Even after multiple rescan some rules showed FAIL even if the auto-remediation are available for them.
@Vincent056 Could you please help me check? Thanks
```
1. Install CO from PR code
2. Create ssb with profiles ocp4-moderate,ocp4-moderate-node, rhcos4-moderate
$ oc compliance bind -N test -S default-auto-apply profile/ocp4-moderate profile/ocp4-moderate-node profile/rhcos4-moderate
Creating ScanSettingBinding test
$ oc get scan
NAME                        PHASE   RESULT
ocp4-moderate               DONE    NON-COMPLIANT
ocp4-moderate-node-master   DONE    NON-COMPLIANT
ocp4-moderate-node-worker   DONE    NON-COMPLIANT
rhcos4-moderate-master      DONE    NON-COMPLIANT
rhcos4-moderate-worker      DONE    NON-COMPLIANT
3. $ oc compliance rerun-now scansettingbinding test
Rerunning scans from 'test': ocp4-moderate, ocp4-moderate-node-master, ocp4-moderate-node-worker, rhcos4-moderate-master, rhcos4-moderate-worker
Re-running scan 'openshift-compliance/ocp4-moderate'
Re-running scan 'openshift-compliance/ocp4-moderate-node-master'
Re-running scan 'openshift-compliance/ocp4-moderate-node-worker'
Re-running scan 'openshift-compliance/rhcos4-moderate-master'
Re-running scan 'openshift-compliance/rhcos4-moderate-worker'
$ oc get suite
NAME   PHASE   RESULT
test   DONE    NON-COMPLIANT
$ oc get ccr -l compliance.openshift.io/inconsistent-check 
No resources found in openshift-compliance namespace.
$ oc get ccr -l compliance.openshift.io/automated-remediation=,compliance.openshift.io/check-status=FAIL | grep service-debug-shell-disabled
rhcos4-moderate-master-service-debug-shell-disabled                    FAIL     medium
rhcos4-moderate-worker-service-debug-shell-disabled                    FAIL     medium
$ oc get ccr -l compliance.openshift.io/automated-remediation=,compliance.openshift.io/check-status=FAIL
NAME                                                                   STATUS   SEVERITY
rhcos4-moderate-master-service-debug-shell-disabled                    FAIL     medium
rhcos4-moderate-master-sysctl-net-core-bpf-jit-harden                  FAIL     medium
rhcos4-moderate-master-sysctl-net-ipv6-conf-all-accept-ra              FAIL     medium
rhcos4-moderate-master-sysctl-net-ipv6-conf-all-accept-redirects       FAIL     medium
rhcos4-moderate-master-sysctl-net-ipv6-conf-default-accept-ra          FAIL     medium
rhcos4-moderate-master-sysctl-net-ipv6-conf-default-accept-redirects   FAIL     medium
rhcos4-moderate-worker-service-debug-shell-disabled                    FAIL     medium
rhcos4-moderate-worker-sysctl-net-core-bpf-jit-harden                  FAIL     medium
rhcos4-moderate-worker-sysctl-net-ipv6-conf-all-accept-ra              FAIL     medium
rhcos4-moderate-worker-sysctl-net-ipv6-conf-all-accept-redirects       FAIL     medium
rhcos4-moderate-worker-sysctl-net-ipv6-conf-default-accept-ra          FAIL     medium
rhcos4-moderate-worker-sysctl-net-ipv6-conf-default-accept-redirects   FAIL     medium
```

**Comment by BhargaviGudi** [comment]
Unable to create ssb with both rhcos4 and ocp4. Observing below error.
```
$ oc compliance bind -N test -S default-auto-apply profile/ocp4-moderate profile/ocp4-moderate-node profile/rhcos4-moderate
Creating ScanSettingBinding test
[bgudi@bgudi-thinkpadt14sgen2i compliance-operator]$ oc get ssb
NAME   STATUS
test   INVALID
$ oc describe ssb test  | tail 
    Last Transition Time:  2024-02-26T08:32:04Z
    Message:               ScanSettingBinding defines multiple products: redhat_enterprise_linux_coreos_4 and redhat_openshift_container_platform_node_4
    Reason:                Invalid
    Status:                False
    Type:                  Ready
  Phase:                   INVALID
Events:
  Type     Reason            Age                    From                    Message
  ----     ------            ----                   ----                    -------
  Warning  MultipleProducts  2m30s (x2 over 2m30s)  scansettingbindingctrl  ScanSettingBinding defines multiple products: redhat_enterprise_linux_coreos_4 and redhat_openshift_container_platform_node_4
```
@Vincent056 Could you please help me check this issue? Thanks

**Comment by Vincent056** [comment]
> Unable to create ssb with both rhcos4 and ocp4. Observing below error.
> 
> ```
> $ oc compliance bind -N test -S default-auto-apply profile/ocp4-moderate profile/ocp4-moderate-node profile/rhcos4-moderate
> Creating ScanSettingBinding test
> [bgudi@bgudi-thinkpadt14sgen2i compliance-operator]$ oc get ssb
> NAME   STATUS
> test   INVALID
> $ oc describe ssb test  | tail 
>     Last Transition Time:  2024-02-26T08:32:04Z
>     Message:               ScanSettingBinding defines multiple products: redhat_enterprise_linux_coreos_4 and redhat_openshift_container_platform_node_4
>     Reason:                Invalid
>     Status:                False
>     Type:                  Ready
>   Phase:                   INVALID
> Events:
>   Type     Reason            Age                    From                    Message
>   ----     ------            ----                   ----                    -------
>   Warning  MultipleProducts  2m30s (x2 over 2m30s)  scansettingbindingctrl  ScanSettingBinding defines multiple products: redhat_enterprise_linux_coreos_4 and redhat_openshift_container_platform_node_4
> ```
> 
> @Vincent056 Could you please help me check this issue? Thanks

could you test without using
`make catalog-deploy CATALOG_IMG=ghcr.io/complianceascode/compliance-operator-catalog:489` this workflow hass issue right now. @BhargaviGudi

**Comment by Vincent056** [comment]
also, `service_debug-shell_disabled failure` is not related to this PR, it is caused by https://github.com/ComplianceAsCode/content/pull/11638.

**Comment by Vincent056** [comment]
> also, `service_debug-shell_disabled failure` is not related to this PR, it is caused by [ComplianceAsCode/content#11638](https://github.com/ComplianceAsCode/content/pull/11638). @BhargaviGudi @xiaojiey

**Review by BhargaviGudi — APPROVED** [review]

File: tests/e2e/serial/main_test.go [hunk 2]
```diff
@@ -247,6 +247,109 @@ func TestSuiteScan(t *testing.T) {
...
+			if s.Name == scan {
+				found = true
+				if s.Phase != compv1alpha1.PhaseDone {
+					t.Fatalf("expected scan %s to be done", scan)
+				}
+				// E2e tests have been flaky because of the file_permissions_var_log_kube_audit
+				// rule. This is because there is a bug in the API-server in old versions of OCP[1][2].
+				// For now, we'll just check that the scan is not inconsistent until we upgrade
+				// to a version that has the fix.
+				// [1]https://bugzilla.redhat.com/show_bug.cgi?id=2001442
```
Comment (rhmdnd) [inline tests/e2e/serial/main_test.go:333]: The errata linked to this was for 4.10, which means it should be fixed there, right?

Did you hit this case in your testing of this e2e test locally? If so, I wonder if something else is causing this since the fix appears to be in 4.10.

**Review by rhmdnd — COMMENTED** [review]
Only one comment on the e2e test. Otherwise, experimenting with this locally works as expected.

Thanks!

File: tests/e2e/serial/main_test.go (hunk 2 above)
Comment (Vincent056) [reply tests/e2e/serial/main_test.go:333]: Yes, this was fixed, I have it flipped the result for the suite

Comment (rhmdnd) [reply tests/e2e/serial/main_test.go:333]: Ok - so does this still manifest in an inconsistent result? I'm wondering if it's safe to remove the inconsistency check to make the test case simpler.

File: pkg/controller/scansettingbinding/scansettingbinding_controller.go [hunk 3]
```diff
@@ -183,27 +182,11 @@ func (r *ReconcileScanSettingBinding) Reconcile(ctx context.Context, request rec
 			}
 		}
 
-		scan, product, err := newCompScanFromBindingProfile(r, instance, profileObj, log)
+		scan, _, err := newCompScanFromBindingProfile(r, instance, profileObj, log)
```
Comment (rhmdnd) [inline pkg/controller/scansettingbinding/scansettingbinding_controller.go:185]: Looks like this is the only place where this is called, and since we're not using `product` anymore we can probably clean it up in another patch.
